Rocq CLI (#19927)
coq/coq: doc/sphinx/practical-tools/coq-commands.rst

.. _therocqcommands:

The Rocq Prover commands
========================

There are several Rocq commands:

+ ``coqide``: a graphical integrated development environment, described
  :ref:`here <coqintegrateddevelopmentenvironment>`.  In addition, there are
  several other IDEs such as Proof General, vsCoq and Coqtail that are not
  included with the Coq installation.

> + ``rocq``: the main entry point for Rocq

Reviewer: Use the longer (full) name when it reads nicely:

```suggestion
+ ``rocq``: the main entry point for the Rocq prover
```

.. _interactive-use:

Interactive use (rocq repl)
---------------------------

> In the interactive mode, also known as the Coq toplevel, users can
> develop their theories and proofs step by step. The Coq toplevel is run

Reviewer: Remove the outdated introduction that this would be how users develop their theories.

```suggestion
The Rocq toplevel (or read-eval-print-loop) is run
```

by the command ``rocq repl`` (equivalently, `rocq top`).

There is also a byte-code toplevel `rocq repl-with-drop` based on an OCaml toplevel.
You can switch to the OCaml toplevel with the command ``Drop.``,

> and come back to the Coq toplevel with the command ``#go;;``.

Reviewer: ```suggestion
and come back to the Rocq toplevel with the command ``#go;;``.
```

.. flag:: Coqtop Exit On Error

   This :term:`flag`, off by default, causes `rocq top` to exit with status code
   ``1`` if a command produces an error instead of recovering from it.

Batch compilation (rocq compile)
--------------------------------

The ``rocq compile`` (equivalently, `rocq c`) command compiles

> a Coq proof script file with a ".v" suffix

Reviewer: ```suggestion
a Rocq proof script file with a ".v" suffix
```

to create a compiled file with a ".vo" suffix.  (See :ref:`compiled-files`.)

> The last component of the filename must be a valid Coq identifier as described in

Reviewer: ```suggestion
The last component of the filename must be a valid Rocq identifier as described in
```

We recommend specifying a :term:`logical path` (which is also the module name)
with the `-R` or the `-Q` options.
Generally we recommend using utilities such as `make` (using `rocq makefile`
to generate the `Makefile`) or `dune` to build Rocq projects.
See :ref:`rocq_makefile` and :ref:`building_dune`.

.. example:: Compiling and loading a single file

   If `foo.v` is in Rocq's current directory, you can use `rocq c foo.v`
   to compile it and then `Require foo.` in your script.  But this
   doesn't scale well for larger projects.

   Generally it's better to define a new module:
   To compile `foo.v` as part of a module `Mod1` that is rooted
   at `.` (i.e. the directory containing `foo.v`), run `rocq c -Q . Mod1 foo.v`.

   To make the module available in `CoqIDE`, include the following line in the
   `_CoqProject` file (see :ref:`rocq_makefile`) in the directory from which you
   start `CoqIDE` or give it as an argument to the ``coqide`` command.
   *<PATH>* is the pathname of the directory containing the module,

>    which can be an absolute path or relative to Coq's current directory.  For now,

Reviewer: ```suggestion
   which can be an absolute path or relative to Rocq's current directory.  For now,
```

`coqrc` start up script
~~~~~~~~~~~~~~~~~~~~~~~

> When Rocq is launched, it can implicitly prepend a startup script to any document
> read by Rocq, whether it is an interactive session or a file to compile.

Reviewer: ```suggestion
When Rocq is launched, it can implicitly prepend a startup script to any document
it reads, whether it is an interactive session or a file to compile.
```
